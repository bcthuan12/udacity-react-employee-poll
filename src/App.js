import "./App.css";
import { handleInitialData } from "./actions/shared";
import { connect } from "react-redux";
import LoginPage from "./components/LoginPage";
import DashboardPage from "./components/DashboardPage";
import { Route, Routes } from "react-router-dom";
import { useEffect } from "react";
import PollCreationPage from "./components/PollCreationPage";
import PollPage from "./components/PollPage";
import LeaderBoardPage from "./components/LeaderBoardPage";

const App = ({ dispatch, loggedIn }) => {
  useEffect(() => {
    dispatch(handleInitialData());
  }, []);
  return (
    <div>
      <Routes>
        <Route path="/login" exact element={<LoginPage />} />
        <Route path="/" element={<DashboardPage />} />
        <Route path="/add" exact element={<PollCreationPage />}></Route>
        <Route path="/questions/:id" element={<PollPage />}></Route>
        <Route path="/leaderboard" exact element={<LeaderBoardPage />}></Route>
      </Routes>
    </div>
  );
};

const mapStateToProps = ({ authorizedUser }) => ({
  loggedIn: !!authorizedUser,
});

export default connect(mapStateToProps)(App);
