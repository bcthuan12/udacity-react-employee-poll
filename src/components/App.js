import { handleInitialData } from "../actions/shared";
import { connect } from "react-redux";
import LoginPage from "./LoginPage";
import DashboardPage from "./DashboardPage";
import { Route, Routes } from "react-router-dom";
import { useEffect } from "react";
import PollCreationPage from "./PollCreationPage";
import PollPage from "./PollPage";
import LeaderBoardPage from "./LeaderBoardPage";
import Interceptor from "./Interceptor";
import NotFoundPage from "./NotFoundPage";

const App = ({ dispatch, loggedIn }) => {
  useEffect(() => {
    dispatch(handleInitialData());
  }, []);
  return (
    <div>
      <Routes>
        <Route path="/404" exact element={<NotFoundPage />} />
        <Route path="/login" exact element={<LoginPage />} />
        <Route
          path="/"
          element={
            <Interceptor>
              <DashboardPage />
            </Interceptor>
          }
        />
        <Route
          path="/add"
          exact
          element={
            <Interceptor>
              <PollCreationPage />
            </Interceptor>
          }
        ></Route>
        <Route
          path="/questions/:id"
          element={
            <Interceptor>
              <PollPage />
            </Interceptor>
          }
        ></Route>
        <Route
          path="/leaderboard"
          exact
          element={
            <Interceptor>
              <LeaderBoardPage />
            </Interceptor>
          }
        ></Route>
      </Routes>
    </div>
  );
};

const mapStateToProps = ({ authorizedUser }) => ({
  loggedIn: !!authorizedUser,
});

export default connect(mapStateToProps)(App);
