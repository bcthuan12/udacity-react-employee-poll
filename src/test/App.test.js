import { render } from "@testing-library/react";
import App from "../App";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import React from "react";
import { store } from "../index";

describe("App", () => {
  test("Should match snapshot", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>,
    );

    expect(component).toMatchSnapshot();
  });
});
