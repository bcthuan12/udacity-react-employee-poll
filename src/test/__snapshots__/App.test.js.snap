// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container"
        >
          <div
            class="justify-content-md-center row"
          >
            <div
              class="text-center col"
            >
              <h1>
                Employee Polls
              </h1>
            </div>
          </div>
          <div
            class="justify-content-md-center row"
          >
            <div
              class="col-sm-6"
            >
              <img
                class="rounded-circle"
                src="login.png"
              />
            </div>
          </div>
          <div
            class="justify-content-md-center row"
          >
            <div
              class="text-center mt-5 col"
            >
              <h3>
                Login
              </h3>
            </div>
          </div>
          <form
            class=""
          >
            <div
              class="mt-3 row"
            >
              <div
                class="col-sm-2"
              >
                <label
                  class="form-label"
                  for="username"
                >
                  Username
                </label>
              </div>
              <div
                class="col-sm-10"
              >
                <input
                  class="form-control"
                  data-testid="inputUserName"
                  id="username"
                  name="username"
                  type="text"
                  value="mtsamis"
                />
              </div>
            </div>
            <div
              class="mt-3 row"
            >
              <div
                class="col-sm-2"
              >
                <label
                  class="form-label"
                  for="password"
                >
                  Password
                </label>
              </div>
              <div
                class="col-sm-10"
              >
                <input
                  class="form-control"
                  data-testid="inputPassword"
                  id="password"
                  name="password"
                  type="password"
                  value="xyz123"
                />
              </div>
            </div>
            <div
              class="mt-3 row"
            >
              <button
                class="btn btn-primary"
                type="submit"
              >
                Login
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container"
      >
        <div
          class="justify-content-md-center row"
        >
          <div
            class="text-center col"
          >
            <h1>
              Employee Polls
            </h1>
          </div>
        </div>
        <div
          class="justify-content-md-center row"
        >
          <div
            class="col-sm-6"
          >
            <img
              class="rounded-circle"
              src="login.png"
            />
          </div>
        </div>
        <div
          class="justify-content-md-center row"
        >
          <div
            class="text-center mt-5 col"
          >
            <h3>
              Login
            </h3>
          </div>
        </div>
        <form
          class=""
        >
          <div
            class="mt-3 row"
          >
            <div
              class="col-sm-2"
            >
              <label
                class="form-label"
                for="username"
              >
                Username
              </label>
            </div>
            <div
              class="col-sm-10"
            >
              <input
                class="form-control"
                data-testid="inputUserName"
                id="username"
                name="username"
                type="text"
                value="mtsamis"
              />
            </div>
          </div>
          <div
            class="mt-3 row"
          >
            <div
              class="col-sm-2"
            >
              <label
                class="form-label"
                for="password"
              >
                Password
              </label>
            </div>
            <div
              class="col-sm-10"
            >
              <input
                class="form-control"
                data-testid="inputPassword"
                id="password"
                name="password"
                type="password"
                value="xyz123"
              />
            </div>
          </div>
          <div
            class="mt-3 row"
          >
            <button
              class="btn btn-primary"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
